diff --git a/filefield_paths.inc b/filefield_paths.inc
index f75563a..68601d7 100644
--- a/filefield_paths.inc
+++ b/filefield_paths.inc
@@ -99,7 +99,7 @@ function filefield_paths_filefield_paths_process_file(ContentEntityInterface $en
       $path = filefield_paths_process_string($settings['file_path']['value'], $token_data, $settings['file_path']['options']);

       $destination = \Drupal::service('stream_wrapper_manager')
-        ->normalizeUri($field_storage->getSetting('uri_scheme') . '://' . $path . DIRECTORY_SEPARATOR . $name);
+        ->normalizeUri($field_storage->getSetting('uri_scheme') . '://' . $path .  '/' . $name);

       // Ensure file uri is no more than 255 characters.
       if (mb_strlen($destination) > 255) {
